packages/cli/src/lib/cmds/merge.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import * as path from 'path';
+import {ParsedArgs} from 'minimist';
+import {util, TwaManifest} from '@bubblewrap/core';
+import {update} from './update';
+
+/**
+ * Updates an existing TWA Project using the `twa-manifest.json`.
+ * @param {string} [args.fieldsToIgnore] the fields that shouldn't be updated.
+ */
+export async function merge(args: ParsedArgs): Promise<boolean> {
+  // If there is nothing to ignore, continue with an empty list.
+  let fieldsToIgnore = [];
+  if (args.ignore) {
+    fieldsToIgnore = args.ignore;
+  }
+  const manifestFile = path.join(process.cwd(), 'twa-manifest.json');
+  const twaManifest = await TwaManifest.fromFile(manifestFile);
+  const webManifestUrl: URL = twaManifest.webManifestUrl!;
+  const webManifest = await util.getWebManifest(webManifestUrl);
+  const newTwaManifest =
+      await TwaManifest.merge(fieldsToIgnore, webManifestUrl, webManifest, twaManifest);
+  // Update the app (args are not relevant in this case, because update's default values
+  // are valid for it. We just send something as an input).
+  await newTwaManifest.saveToFile(path.join(process.cwd(), 'twa-manifest.json'));
+  await update(args);
```

Comment:
If `args` aren't relevant, could we just pass null here?

Reply:
I tried that, it isn't working. it says: "Argument of type 'null' is not assignable to parameter of type 'ParsedArgs'."
same with undefined

Reply:
Looking at the update implementation, the bits that seem to matter from that code are:

```ts
const newVersionInfo = await updateVersions(twaManifest, args.appVersionName, prompt);
const twaGenerator = new TwaGenerator();
await twaGenerator.removeTwaProject(targetDirectory);
await generateTwaProject(prompt, twaGenerator, targetDirectory, twaManifest);
```
The other bits seem to mainly related to handling args parameters (which we don't care about)

Reply:
it sends it to `updateVersions` and then we get the following code: 
```// If a version was passed as parameter, use it.
  if (appVersionNameArg) {
    return {
      appVersionCode: appVersionCode,
      appVersionName: appVersionNameArg,
    };
  }

  // Otherwise, try to upgrade automatically with the versionCode.
  if (twaManifest.appVersionName === previousAppVersionCode.toString()) {
    return {
      appVersionCode: appVersionCode,
      appVersionName: appVersionCode.toString(),
    };
  }

  // If not not possible, ask the user to input a new version.
  const appVersionName = await prompt.promptInput(
      messages.promptNewAppVersionName,
      null,
      createValidateString(1),
  );
```

in all of the situations we have a fine solution, and when I run it it went perfectly

Reply:
Looking at the other cmds, I don't think we have an instance of once command invoking another command directly. One reason for this is that it makes it easier to reason when changing to code in one command - By invoking `update` from `merge`,  we now need to be careful when changing `update`, like adding a new flag to the command, as it may affect `merge`.  But this relationship is not obvious and folks working on the project are likely to forget.

Examples of similar problems where we created shared code: 
 - the shared code between `init` and `update` has been extracted to`shared.ts`. 
 - the shared code between `validate` and `build` has been extracted to `pwaValidationHelper.ts`.

Reply:
So should I move this function to `shared.ts`?
I don't see the difference this will make, we still have one function which both of the commands use, and we need to be careful when doing changes in one of them

Reply:
Yes, please. It doesn't make a functional difference, but the code becomes easier to maintain. As a developer who's not familiar with the project (or even ourselves modifying the code in the future), it is not immediately obvious that one command would invoke the other (eg: `merge` invoke `update`). But when you see a command invoking a function in another file (especially if this file is `shared.ts`), it's pretty reasonable to assume that the function is used in multiple places, and that any changes should have this in mind.